### DISCUSSION TITLE
Configure Home-Idp-Discovery Config with Multi Realms

### DISCUSSION THREAD

**Discussion by khanhcd92** [sventorben/keycloak-home-idp-discovery#337] [Q&A]
Could we use this extension for multi reamls ?

**Comment by sventorben** [comment]
I need more context info to understand what you want to achieve and toanswer your question.

**Reply by khanhcd92** [reply]
I would like to use KeyCloak to build authentication and authorization for multi company. I found this extention.
Example: I have 2 company. Each company has a Idp(AzureAD). I want to use realm for them. 
- Realm A -> Idp(AzureAd)
- Realm B -> Idp(AzureAd)

So I use this extention for my purpose, don't I ? And How?
If we could use it, please help me to setup it.

@sventorben

**Reply by sventorben** [reply]
I'm sorry if I have to ask further. I still don't understand the problem. If both companies have their own realm and each of these companies only has one identity provider, why don't you always redirect them to exactly that one identity provider?
At what point is it unclear which identity provider should be redirected to, so that this extension is even necessary?

**Comment by oferchen** [comment]
I think I understand the issue as I'm trying to resolve the same thing.
I have multiple realms let's call it realm1 and realm2 because I need a flat hierarchy to use this extension I created a broker realm and mapped realm1 and realm2 as identity providers to it. Now I have a single realm with access to the identities from the two other realms and I'm trying to configure home idp dicovery to redirect the users according to the realm home.idp.discovery.domains attribute.
I think the flat hierarchy allows for more flexibility in terms of which idp you log on to.
